hc_core_api_c_binding/src/lib.rs

```diff
@@ -0,0 +1,112 @@
+#![deny(warnings)]
+extern crate hc_agent;
+extern crate hc_core;
+extern crate hc_core_api;
+extern crate hc_dna;
+
+use hc_core::context::Context;
+use hc_core_api::Holochain;
+use hc_dna::Dna;
+use std::sync::Arc;
+
+use hc_agent::Agent;
+use hc_core::logger::Logger;
+use hc_core::persister::SimplePersister;
+use std::ffi::{CStr, CString};
+use std::os::raw::c_char;
+use std::sync::Mutex;
+
+#[derive(Clone, Debug)]
+struct NullLogger {}
+
+impl Logger for NullLogger {
+    fn log(&mut self, _msg: String) {}
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn hc_new(ptr: *mut Dna) -> *mut Holochain {
+    let agent = Agent::from_string("c_bob");
```

Comment:
rust ffi docs recommend catching panics as they result in undefined behavior... although panics seem pretty bad anyway, so may not be a big issue?

https://doc.rust-lang.org/book/first-edition/ffi.html#ffi-and-panics